### PR TITLE
Remove hard-coding of teaser formats

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Allows the number/wording to be changed.

https://github.com/elifesciences/issues/issues/5416

(I could have sworn it already worked like this, maybe we just talked about it and didn't do it.)

### CODE REVIEW

File: source/_patterns/04-pages/home.json [hunk 1]
```diff
@@ -377,8 +377,7 @@
               }
             },
             "formats": {
-              "html": true,
-              "pdf": true
+              "list": ["Executable", "PDF"]
```
Comment (thewilkybarkid) [inline source/_patterns/04-pages/home.json]: 👋

File: source/_patterns/01-molecules/teasers/teaser.yaml [hunk 2]
```diff
@@ -85,12 +85,14 @@ properties:
...
         properties:
-          html:
-            type: boolean
-          pdf:
-            type: boolean
+          list:
+            type: array
+            minItems: 1
+            item:
+              type: string
```
Comment (giorgiosironi) [inline source/_patterns/01-molecules/teasers/teaser.yaml]: not super future-proof if something like a description gets added (or any metadata). May be good for now given how relatively easy this schema can change.

File: source/_patterns/04-pages/home.json (hunk 1 above)
Comment (giorgiosironi) [reply source/_patterns/04-pages/home.json]: shouldn't this also have `HTML`?

Comment (giorgiosironi) [reply source/_patterns/04-pages/home.json]: I know this is just an example, but is the uppercase implemented via CSS?

Comment (giorgiosironi) [reply source/_patterns/04-pages/home.json]: > shouldn't this also have HTML?

Ah re-read the whole issue, there's... not enough space for that? Whatever, the example is intended.

**Pull request merged**